DataViews Pages: Dynamic Frame title as per active view for Pages and Templates editor pages

Souptik2001 opened the pull request:
<!-- Thanks for contributing to Gutenberg! Please follow the Gutenberg Contributing Guidelines:
https://github.com/WordPress/gutenberg/blob/trunk/CONTRIBUTING.md -->

## What?
<!-- In a few words, what is the PR actually doing? -->

In DataList views for pages, add a subtitle to display the activeView.

## Why?
<!-- Why is this PR necessary? What problem is it solving? Reference any existing previous issue(s) or PR(s), but please add a short summary here, too -->

Closes: #56199 

This requirement is first mentioned in the above issue.

But a more refined requirement can be found in this comment - https://github.com/WordPress/gutenberg/pull/56241#issuecomment-1816762258

Just looking at the whitespace of the DataList Views of pages, it might be a little unclear what activeView you are on. That's why we can add a sub-title below the title "Pages" to display the activeView over there.

## How?
<!-- How is your PR addressing the issue at hand? What are the implementation details? -->

This PR adds a sub-title for DataViews screen for pages, which displays the activeView.

## Testing Instructions
<!-- Please include step by step instructions on how to test this PR. -->
<!-- 1. Open a post or page. -->
<!-- 2. Insert a heading block. -->
<!-- 3. etc. -->

Please see the below screencast section or refer the steps below.

- Open the site-editor.
- Go to "Pages".
- Now switch the active view by clicking on the left sidebar.
- As you change the views you will see that the subtitle under the title "Pages" also changes accordingly.

### Testing Instructions for Keyboard
<!-- How can you test the changes by using the keyboard only? Please note, this is required for PRs that change the user interface (UI). This ensures the PR can be tested for any possible accessibility regressions. -->

N/A

## Screenshots or screencast <!-- if applicable -->


https://github.com/WordPress/gutenberg/assets/39427067/1019c127-dcf1-42db-9b99-0159163b54b6

**OLD VIDEO:**

https://github.com/WordPress/gutenberg/assets/39427067/94712b02-8a93-4527-a957-7c93f2816073

Souptik2001 committed:
feat: Add frame subtitle for page active view

Signed-off-by: Souptik Datta <souptikdatta2001@gmail.com>

Souptik2001 committed:
Merge branch 'WordPress:trunk' into update/dataviews-add-frame-subtitle-for-active-view

jameskoster commented:
Thank you for opening the PR :) 

I think it probably makes sense to use the main title, rather than a subtitle, at least initially. This would align with the Pattern management page, and leave the subtitle free for additional context to be added later if needed.

What do you think?

Souptik2001 commented:
> Thank you for opening the PR :)
> 
> I think it probably makes sense to use the main title, rather than a subtitle, at least initially. This would align with the Pattern management page, and leave the subtitle free for additional context to be added later if needed.
> 
> What do you think?

@jameskoster Yes I align with your thought! That would be better!

I made it like this as per this comment - https://github.com/WordPress/gutenberg/pull/56241#issuecomment-1816762258

Just taking a confirmation, that should I change it?

Souptik2001 committed:
Merge branch 'trunk' into update/dataviews-add-frame-subtitle-for-active-view

Souptik2001 committed:
refactor: Movie viewTitle from subtitle to title of page

Signed-off-by: Souptik Datta <souptikdatta2001@gmail.com>

Souptik2001 commented:
Thanks @jameskoster for confirming!

I have made the update! Thanks!

jameskoster commented:
This works well for me. I think it would make sense to apply the same treatment to the templates page. What do you think?

Souptik2001 committed:
feat: Show active view name on templates frame title

Signed-off-by: Souptik Datta <souptikdatta2001@gmail.com>

Souptik2001 commented:
Thanks @jameskoster! Yes we can also update templates to have the same behaviour! 👍 

I have made that change!
I have also updated the video in the description, you can check now for both page and template.
Also I have updated the PR title to make it more appropriate.

jameskoster commented:
Thanks for updating the Templates page :)

There is a quirk (as I think you mentioned on the issue): if you reset the filters on a bundled page, the title can be misleading. E.g. Go to trashed pages, remove the status filter, and you're no longer viewing trashed pages:

<img width="1152" alt="Screenshot 2024-05-30 at 11 09 31" src="https://github.com/WordPress/gutenberg/assets/846565/a9360e15-bf6d-407c-8e65-0d4ea4f80c8d">

This will be fixed when https://github.com/WordPress/gutenberg/issues/60468 is addressed. So the question here is whether we should include this change in the upcoming 6.6 release, or whether we should consider #60468 a blocker.

The main argument for including it is to establish alignment with the Patterns page, where the frame title matches the menu item:

<img width="1006" alt="Screenshot 2024-05-30 at 11 16 50" src="https://github.com/WordPress/gutenberg/assets/846565/f7802ee9-1953-4162-bb90-1823eb5ecb81">


I'm keen to gather thoughts from others on that cc @WordPress/gutenberg-design.

t-hamano commented:
> There is a quirk (as I think you mentioned on the issue): if you reset the filters on a bundled page, the title can be misleading. E.g. Go to trashed pages, remove the status filter, and you're no longer viewing trashed pages:

To avoid this discrepancy, how about excluding the sidebar items from the data view filter?

For example, on a page, if you select "Draft" in the sidebar, the data view will only list posts with a Draft status. However, it does not add any filters and we cannot reset the post status in the DataViews.

jameskoster commented:
@t-hamano that sounds like the first point in https://github.com/WordPress/gutenberg/issues/60468. I think the question is whether or not we consider that a blocker to this PR.

t-hamano commented:
Ah, I didn't read that issue thoroughly enough 😅 Personally, I would lean towards not including this PR in 6.6 and moving the discussion forward in #60468.

Souptik2001 commented:
@jameskoster @t-hamano I can get that task rolling. 🙂 

If I understood correctly, for this first version we just want a visual feedback of whether the view is editor or not right(as per the design is provided)? Like the "Save view" feature, etc. will be introduced later, correct?

If you give the go-ahead I can work on that one. 🙂 
Thanks!

jameskoster commented:
Yup, I think the first three tasks in https://github.com/WordPress/gutenberg/issues/60468 are related, and can potentially be tackled in a single PR. The fourth and fifth tasks can come later 👍

Souptik2001 commented:
Thanks @jameskoster! I have raised a PR for first three points of #60468 !
